docs/getting_started/middleware.md

```diff
@@ -0,0 +1,145 @@
+# Gosoline middleware
+
+We saw in the [Kernel](kernel.md) section, that the _kernel.Kernel_ interface offers a method to add middleware to a kernel:
+
+[embedmd]:# (../../pkg/kernel/kernel.go /AddMiddleware\(/ /\)/)
+```go
+AddMiddleware(middleware Middleware, position Position)
+```
+
+The position of a middleware is one of two values:
+
+[embedmd]:# (../../pkg/kernel/middleware.go /type Position/ /\)/)
+```go
+type Position string
+
+const (
+	PositionBeginning Position = "beginning"
+	PositionEnd       Position = "end"
+)
+```
+
+Lastly, a _Middleware_ is a function which takes in the following parameters:
+
+[embedmd]:# (../../pkg/kernel/middleware.go /type \(/ /\n\)/)
+```go
+type (
+	Middleware func(ctx context.Context, config cfg.Config, logger log.Logger, next Handler) Handler
+	Handler    func()
+)
+```
+
+With these notions, we are ready for an example.
```

Comment:
Hmm, but even the example doesn't tell me what the real use-case for this is besides printing some strings. What middlewares do we already have, for what can I use them?